wire-ios-request-strategy/Sources/E2EIdentity/ACME/AcmeClient.swift

```diff
@@ -44,15 +46,77 @@ public class AcmeClient: NSObject, AcmeClientInterface {
 
         let path = "https://acme.\(domain)/acme/defaultteams/directory"
 
-        let request = ZMTransportRequest(getFromPath: path, apiVersion: 0)
-        let result = try await httpClient.send(request)
+        guard let url = URL(string: path) else {
+            throw NetworkError.invalidRequest
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = Constant.HTTPMethod.get
+        let (data, _) = try await httpClient.send(request)
+
+        return data
+
+    }
+
+    public func getACMENonce(path: String) async throws -> String {
+
+        guard let url = URL(string: path) else {
+            throw NetworkError.invalidRequest
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = Constant.HTTPMethod.head
 
-        guard let data = result.rawData else {
+        let (_, response) = try await httpClient.send(request)
+
+        guard let httpResponse = response as? HTTPURLResponse,
+              let replayNonce = httpResponse.value(forHTTPHeaderField: Constant.Header.replayNonce) else {
             throw NetworkError.invalidResponse
         }
 
-        return data
+        return replayNonce
+
+    }
+
+    public func sendACMERequest(path: String, requestBody: Data) async throws -> ACMEResponse {
+        guard let url = URL(string: path) else {
+            throw NetworkError.invalidRequest
+        }
+        var request = URLRequest(url: url)
+        request.httpMethod = Constant.HTTPMethod.post
+        request.setValue(Constant.ContentType.joseJson, forHTTPHeaderField: "Content-Type")
+        request.httpBody = requestBody
+
+        let (data, response) = try await httpClient.send(request)
+
+        guard
+            let httpResponse = response as? HTTPURLResponse,
+            let replayNonce = httpResponse.value(forHTTPHeaderField: Constant.Header.replayNonce),
+            let location = httpResponse.value(forHTTPHeaderField: Constant.Header.location)
+        else {
+            throw NetworkError.invalidResponse
+        }
+
+        return ACMEResponse(nonce: replayNonce, location: location, response: data)
+
+    }
+
+}
+
+private enum Constant {
+
+    enum HTTPMethod {
+        static let get = "GET"
+        static let post = "POST"
+        static let head = "HEAD"
+    }
+
+    enum Header {
+        static let replayNonce = "Replay-Nonce"
+        static let location = "location"
+    }
 
+    enum ContentType {
+        static let json = "application/json"
+        static let joseJson = "application/jose+json"
```

Comment:
Suggestion, maybe you like it
```suggestion
        static let joseAndJson = "application/jose+json"
```